src/main/scala/io/qbeast/spark/utils/QbeastUtils.scala

```diff
@@ -28,39 +30,24 @@ import org.apache.spark.sql.DataFrame
 object QbeastUtils extends Logging {
 
   /**
-   * Compute the histogram for a given column
-   *
-   * Since computing the histogram can be expensive, this method is used outside the indexing
-   * process.
-   *
-   * It outputs the histogram of the column as format [bin1, bin2, bin3, ...] Number of bins by
-   * default is 50
-   *
-   * For example:
-   *
-   * val qbeastTable = QbeastTable.forPath(spark, "path")
-   *
-   * val histogram =qbeastTable.computeHistogramForColumn(df, "column")
-   *
-   * df.write.format("qbeast").option("columnsToIndex",
-   * "column:histogram").option("columnStats",histogram).save()
+   * Compute the quantiles for a given column of type String
    *
    * @param df
+   *   the DataFrame
    * @param columnName
+   *   the name of the column
    */
-  def computeHistogramForColumn(df: DataFrame, columnName: String, numBins: Int = 50): String = {
+  private def computeQuantilesForStringColumn(
```

Comment:
We are using two approaches to compute the quantiles for string and numeric columns; this may need to be revised.— `MultiDimClusteringFunctions.range_partition_id` probably works for all data types.

Reply:
preferred to use a Spark method embedded in the df for numeric.